src/main/java/org/jabref/gui/contentselector/ContentSelectorDialogViewModel.java

```diff
@@ -0,0 +1,186 @@
+package org.jabref.gui.contentselector;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import org.jabref.gui.BasePanel;
+import org.jabref.gui.DialogService;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.metadata.ContentSelector;
+import org.jabref.model.metadata.MetaData;
+
+import static com.google.common.collect.ImmutableList.of;
+
+class ContentSelectorDialogViewModel {
+
+    private static final List<String> DEFAULT_FIELD_NAMES = of(FieldName.AUTHOR, FieldName.JOURNAL, FieldName.KEYWORDS, FieldName.PUBLISHER);
+
+    private final MetaData metaData;
+    private final DialogService dialogService;
+    private final BasePanel basePanel;
+    private final Map<String, List<String>> fieldNameKeywordsMap = new HashMap<>();
+
+    private ObservableList<String> fieldNames = FXCollections.observableArrayList();
```

Comment:
Convert this to a `LIstProperty<String> fieldNamesProperty = new SimpleListProperty(FXCollections.observableArrayList())`

Reply:
Fixed e4f3ea3